tests/functional/ovndbcluster_controller_test.go

```diff
@@ -27,11 +28,152 @@ import (
+
+					_ = th.CreateNetworkAttachmentDefinition(types.NamespacedName{Namespace: namespace, Name: "internalapi"})
+					dbs = CreateOVNDBClusters(namespace, map[string][]string{namespace + "/internalapi": {"10.0.0.1"}}, 3)
+					OVNSBDBClusterName = types.NamespacedName{Name: dbs[1].Name, Namespace: dbs[1].Namespace}
+					cluster = GetOVNDBCluster(OVNSBDBClusterName)
+					clusterName = OVNSBDBClusterName
+					var FullDNSEntryName = DNSEntryName + "." + cluster.Namespace + ".svc"
+
+					// Check that DNSData info has been created with correct IP (10.0.0.1)
+					Eventually(func(g Gomega) {
+						// Get DNSData
+						dnsEntry := GetDNSData(types.NamespacedName{Name: "dns-ovsdbserver-sb-0", Namespace: cluster.Namespace})
+						// DNSData contains Hosts (an array), this array will contain
+						// ovsdbserver-sb-0.ns.svc and ovsdbserver-sb.ns.svc, need to get
+						// in which index is located the one which the flavor is currently testing (DNS Cname or DNS)
+						index := -1
+						for i, element := range dnsEntry.Spec.Hosts {
+							for _, el_host := range element.Hostnames {
+								if el_host == FullDNSEntryName {
+									index = i
+								}
+							}
+						}
+						g.Expect(index).To(BeNumerically(">=", 0))
+						g.Expect(dnsEntry.Spec.Hosts[index].IP).Should(Equal("10.0.0.1"))
+					}).Should(Succeed())
+
+					// Modify pod IP section
+					pod := GetPod(types.NamespacedName{Name: "ovsdbserver-sb-0", Namespace: cluster.Namespace})
+					// Create new pod NAD and add it to POD
+					var netStatus []networkv1.NetworkStatus
+					var IPString []string
+					IPString = append(IPString, "10.0.0.10")
+					netStatus = append(
+						netStatus,
+						networkv1.NetworkStatus{
+							Name: cluster.Namespace + "/internalapi",
+							IPs:  IPString,
+						},
+					)
+					netStatusAnnotation, err := json.Marshal(netStatus)
+					Expect(err).NotTo(HaveOccurred())
+					pod.Annotations[networkv1.NetworkStatusAnnot] = string(netStatusAnnotation)
+					UpdatePod(pod)
+					// End modify pod IP section
+
+					// Call reconcile loop
+					Eventually(func(g Gomega) {
+						c := GetOVNDBCluster(clusterName)
+						// Change something just to call reconcile loop
+						*&c.Spec.ElectionTimer = 11000
+						g.Expect(k8sClient.Update(ctx, c)).Should(Succeed())
+					}).Should(Succeed())
+
+					// Check that DNSData info has been modified with correct IP (10.0.0.10)
+					Eventually(func(g Gomega) {
+						dnsEntry := GetDNSData(types.NamespacedName{Name: "dns-ovsdbserver-sb-0", Namespace: cluster.Namespace})
+						// DNSData contains Hosts (an array), this array will contain
+						// ovsdbserver-sb-0.ns.svc and ovsdbserver-sb.ns.svc, need to get
+						// in which index is located the one which the flavor is currently testing (DNS Cname or DNS)
+						index := -1
+						for i, element := range dnsEntry.Spec.Hosts {
+							for _, el_host := range element.Hostnames {
+								if el_host == FullDNSEntryName {
+									index = i
```

Comment:
you repeat the same code to find the hostname here; consider moving into a helper function

Reply:
Done